compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1175,8 +1175,22 @@ class Namer { typer: Typer =>
             if mbr.isType then
               val forwarderName = checkNoConflict(alias.toTypeName, isPrivate = false, span)
               var target = pathType.select(sym)
-              if target.typeParams.nonEmpty then
-                target = target.EtaExpand(target.typeParams)
+              val tparams = target.typeParams
```

Comment:
Should we integrate that logic with eta expand? Or maybe make a variant of it. Meybe it's better to concentrate this quite involved logic in TypeApplications, instead of breaking it out here.